### PR TITLE
Add dockerfile template

### PR DESCRIPTION
**Pull request by langchain-infra**
Description: This pr adds a dockerfile to an app template for easy containerization of your production application.
Dependencies: None

### CODE REVIEW

File: libs/cli/langchain_cli/project_template/Dockerfile [hunk 1]
```diff
@@ -0,0 +1,21 @@
+FROM python:3.11-slim
+
+RUN pip install poetry==1.5.1
```
Comment (efriis) [inline libs/cli/langchain_cli/project_template/Dockerfile:3]: Let's do 1.6.1 to match other libs (can confirm that's right number)

File: libs/cli/langchain_cli/project_template/Dockerfile [hunk 2]
```diff
@@ -0,0 +1,21 @@
...
+
+WORKDIR /code
+
+COPY ./pyproject.toml ./poetry.lock* ./
+
+COPY ./packages ./packages
+
+RUN poetry install --no-interaction --no-ansi --no-root
+
+COPY ./ ./
```
Comment (efriis) [inline libs/cli/langchain_cli/project_template/Dockerfile:15]: Should we just do app here?

File: libs/cli/langchain_cli/project_template/Dockerfile (hunk 1 above)
Comment (langchain-infra) [reply libs/cli/langchain_cli/project_template/Dockerfile]: fixed

File: libs/cli/langchain_cli/project_template/Dockerfile (hunk 2 above)
Comment (langchain-infra) [reply libs/cli/langchain_cli/project_template/Dockerfile]: u right, fixed

File: libs/cli/langchain_cli/project_template/Dockerfile [hunk 3]
```diff
@@ -0,0 +1,21 @@
...
+
+COPY ./pyproject.toml ./poetry.lock* ./
+
+COPY ./packages ./packages
+
+RUN poetry install  --no-interaction --no-ansi --no-root
+
+COPY ./app ./app
+
+RUN poetry install --no-interaction --no-ansi
```
Comment (jakerachleff) [inline libs/cli/langchain_cli/project_template/Dockerfile:17]: @efriis can templates decide to mount any data/files/etc? And is that something we'll have to handle in the docker file (or leave a comment on where you would do that?)

Comment (efriis) [reply libs/cli/langchain_cli/project_template/Dockerfile:17]: In the current setup, all data/files would have to be in the template dir, which gets installed at packages/x. So should be included by default.

We should figure out which concrete use cases we'd want to support with anything else. The main ones people seem to be including currently are 1) jupyter notebooks to show how to use remoterunnable, and 2) data to ingest (either via notebook, script, or commented out code in template). None of these feel like they'd need to move out of packages/x to me, and would love to hear your thoughts!

**Review by efriis — APPROVED** [review]
@langchain-infra lmk what you think of the readme update! All works for me locally :)

**Comment by langchain-infra** [comment]
> @langchain-infra lmk what you think of the readme update! All works for me locally :)

@efriis 
Looks great

**Pull request merged**